### PR TITLE
feat(plugin): created slack plugin with tutorial

### PR DESCRIPTION
**Pull request by mhintz-clickup**
Using this plugin will allow you to receive notifications from the Access app in Slack.
This is intended to be utilized with a Slack Application (https://api.slack.com/apps/).
- Included in the Readme is proper permissions needed and which token to use.

### CODE REVIEW

**Comment by mhintz-clickup** [comment]
closing as meant to get this reviewed by a colleague first.

**Comment by jonathanhle** [comment]
Honestly, we wrote something very similar based on the Discord example, so it's not surprising :). See https://github.com/discord/access/pull/199.  

Let's cross review and join up on the same example.  See you on the Discord Server.

File: examples/plugins/slack/Readme.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- users:read.email  # Must Have
+
+
+3. Create a slack channel and add the Application.
+- Should be under `Integrations` tab within Channel management.
+
+4. Add your slack bot token to the .env.production file and the Channel Name.
+```
+SLACK_BOT_TOKEN=<your-slack-bot-token>
+BASE_URL=<your-base-url>
```
Comment (jonathanhle) [inline examples/plugins/slack/Readme.md:23]: There's already going to be a Client Origin Env Var defined, so why not just use that?

File: examples/plugins/slack/Readme.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+``` 
+
+5. Adjust your Dockerfile and add in the below; Can put it just under the Final build step and above the `ENV ..` lines.
+```
+# Install the plugins
```
Comment (jonathanhle) [inline examples/plugins/slack/Readme.md:28]: Woohoo!

File: examples/plugins/slack/notifications.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        get_slack_client._instance = SlackNotifier(
+            token=token,
+            default_channel=DEFAULT_SLACK_CHANNEL,
+        )
+    return get_slack_client._instance
+
+
+def get_base_url() -> str:
+    # First check for explicit BASE_URL environment variable
+    base_url = os.environ.get("BASE_URL")
```
Comment (jonathanhle) [inline examples/plugins/slack/notifications.py:93]: reuse CLIENT_ORIGIN_URL ?

File: examples/plugins/slack/notifications.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+    requester_message = (
+        f"Request for access to {group.name} has been {access_request.status.lower()}.\n\n"
+        f"[View request]({access_request_url})"
+    )
+
+    # Send the message to both the default channel and the requester
+    logger.info(f"Requester message: {requester_message}")
+    slack = get_slack_client()
+    slack.send_message(message=requester_message)  # Sends to default channel
```
Comment (jonathanhle) [inline examples/plugins/slack/notifications.py:176]: this will fail if a default channel isn't configure.  i'd really prefer this be configurable.

**Comment by mhintz-clickup** [comment]
closing in favor of the other implementation

**Pull request closed without merging**